service/systemd/service_test.go

```diff
@@ -93,16 +85,43 @@ func (s *initSystemSuite) SetUpTest(c *gc.C) {
 		Desc:      "juju agent for " + tagStr,
 		ExecStart: jujud + " " + tagStr,
 	}
-	s.service = s.newService(c)
+}
+
+func (s *initSystemSuite) patch(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
 
-	// Reset any incidental calls.
-	s.stub.ResetCalls()
+	s.dBus = NewMockDBusAPI(ctrl)
+	s.ch = systemd.PatchNewChan(s)
+	s.fops = systemd.PatchFileOps(s, ctrl)
+	s.exec = systemd.PatchExec(s, ctrl)
+
+	return ctrl
 }
 
 func (s *initSystemSuite) newService(c *gc.C) *systemd.Service {
-	service, err := systemd.NewService(s.name, s.conf, s.dataDir)
+	// If we have a mock for the DBusAPI, use it in the factory method passed
+	// into the call to NewService. Otherwise use the default.
+	fac := systemd.NewDBusAPI
```

Comment:
In what situation would we not have a mock dbus? Shouldn't that be an error? Surely we never want to use the system one?

Reply:
Some of the tests don't use the mocks mocks/patching.

You are right are right though; this should be a method of the same signature returning an error so as to fail the test when improperly arranged.